### PR TITLE
Add more tests to Rule_tests

### PR DESCRIPTION
**Pull request by joseemds**
* Add cases there were missing in the existing tests
* Add two cases of tests using the Rule API

### CODE REVIEW

**Review by davesnx — COMMENTED** [review]
Left a bunch of comments, but overall all comments make sense!

File: packages/reason-css-parser/test/Rules_test.re [hunk 1]
```diff
@@ -7,11 +7,46 @@ describe("Data monad", ({test, _}) => {
...
+      | (Ok(","), [COMMA]) => ()
+      | _ => failwith("Should be (Ok(,), [COMMA])")
+      }
+    });
+    test("return with a string", _ => {
+      switch (return(Ok(","), [COMMA])) {
+      | (Ok(","), [COMMA]) => ()
+      | _ => failwith("Should be (Ok(,), [COMMA])")
+      }
+    });
```
Comment (davesnx) [inline packages/reason-css-parser/test/Rules_test.re:34]: Having it twice seems like an error

Comment (joseemds) [reply packages/reason-css-parser/test/Rules_test.re:34]: It is, thanks!

File: packages/reason-css-parser/test/Rules_test.re [hunk 2]
```diff
@@ -7,11 +7,46 @@ describe("Data monad", ({test, _}) => {
...
+    test("return with more then one token as inputs", _ => {
+      switch (return(Ok(1), [IDENT("decl"), COLON, IDENT("value")])) {
+      | (Ok(1), [IDENT("decl"), COLON, IDENT("value")]) => ()
+      | _ =>
+        failwith(
+          "Should be Ok(1), [IDENT(\"decl\"), COLON, IDENT(\"value\")]",
+        )
+      }
+    });
```
[the remark this answers is not in the document]
Comment (davesnx) [reply packages/reason-css-parser/test/Rules_test.re]: It alright

**Pull request merged**